mobile/components/views/Attractions/Search/Search.tsx

```diff
@@ -0,0 +1,159 @@
+      );
+    });
+    setAttractions(filteredAttractions);
+  }, [filters]);
+
+  const handleSetFilters = (filtersProps: Filters) => {
+    setFilterModalVisible(false);
+    setFilters(filtersProps);
+  };
+
+  const handleAttractionClick = (attraction: Attraction) => {
+    // setAttractionModalVisible(true);
+    setAttractionDetails(attraction);
+    navigation.navigate('AttractionDetailScreen', { id: attraction.id });
+
+    console.log('Clicked Attraction:', attraction);
+  };
+
+  const RightSwipeActions = (item: Attraction) => {
+    return (
+      <View style={{ alignItems: 'flex-end', flex: 1, marginTop: 40, padding: 10 }} >
+        <Button title='Add to favorite' onPress={() => {
+          console.log('add to favorite: ', item);
+          // swip.close();
+        }} />
+      </View>
+    );
+  };
+
+  const LeftSwipeActions = (item: Attraction) => {
+    return (
+      <View style={{ alignItems: 'flex-start', flex: 1, marginTop: 40, padding: 10 }} >
+        <Button title='Add to trip' color='green' onPress={() => console.log('add to trip: ', item)} />
+      </View>
+    );
+  };
+
+
+  return (
+    <>
+      <Button title="Filters" onPress={() => setFilterModalVisible(true)} />
+      <FiltersModal
+        visible={filterModalVisible}
+        onClose={() => setFilterModalVisible(false)}
+        onSave={handleSetFilters}
+        attractionList={userAttractionsData}
+        radiuses={radiuses}
+      />
+
+      <AttractionModal
+        visible={attractionModalVisible}
+        attraction={attractionDetails as Attraction}
+        onClose={() => setAttractionModalVisible(false)}
+      />
+
+      <View style={styles.attractionsGrid}>
+
+        {loading && (
+          <Text>Loading data</Text>
+        )}
+
+        {!loading && attractions.length === 0 && (
+          <View>
+            <Image
+              source={require('./../../../../resources/images/we_dont_have.png')}
+              style={{ width: 200, height: 200, margin: 20, padding: 10 }}
+            />
+          </View>
+        )}
+
+        {!loading && attractions.length > 0 && (
+          <FlatList
+            style={attractionSerchStyles.attractionsList}
+            data={attractions}
+            keyExtractor={(item, index) => index.toString()}
+            renderItem={({ item: attraction }) => (
+              <TouchableOpacity onPress={() => handleAttractionClick(attraction)}>
+                <Swipeable
+                  renderRightActions={() => RightSwipeActions(attraction)}
+                  renderLeftActions={() => LeftSwipeActions(attraction)}
+                >
+                  <View>
+                    <AttractionTile
+                      attraction={{
+                        place_name: attraction.name,
+                        place_description: attraction.description,
+                        place_category: attraction.category,
+                        place_rating: attraction.rating.toFixed(1)
+                      }}
```

Comment:
Also here, diff is buggy so I cant reach whole component, but still remember that if this attractions are fetched somehow without earlier check if its not `undefined` errors will still occure.

Reply:
I check whether the attraction has been fetched and determine this via the loading flag.